lib/package.rb

```diff
@@ -276,19 +276,39 @@ def self.system(*args, **opt_args)
       env = @crew_env_options_hash
     end
 
-    # after removing the env hash, all remaining args must be command args
-    cmd_args = args
-
-    # add -j arg to build commands
-    if args.size == 1
-      # involve a shell if the command is passed in one single string
-      cmd_args = ['bash', '-c', cmd_args[0].sub(/^(make)\b/, "\\1 -j#{CREW_NPROC}")]
-    elsif cmd_args[0] == 'make'
-      cmd_args.insert(1, "-j#{CREW_NPROC}")
+    cmd_args        = args  # after removing the env hash, all remaining args must be command args
+    make_threads    = CREW_NPROC
+    modded_make_cmd = false
+
+    # append -j placeholder to `make` commands only if '-j#' does not exist
+    unless cmd_args.grep(/-j[[:space:]]?[0-9]+/).any?
+      if cmd_args.size == 1
+        # involve a shell if the command is passed in one single string
+        cmd_args        = ['bash', '-c', cmd_args[0].sub(/^(make)\b/, "\\1 <<<CREW_NPROC>>>")]
+        modded_make_cmd = true
+      elsif cmd_args[0] == 'make'
+        cmd_args.insert(1, '<<<CREW_NPROC>>>')
+        modded_make_cmd = true
+      end
     end
 
     begin
-      Kernel.system(env, *cmd_args, **opt_args)
+      if modded_make_cmd
+        # replace placeholder with '-j#' arg and execute the actual command
+        Kernel.system(env, *cmd_args.map { |arg| arg.sub('<<<CREW_NPROC>>>', "-j#{make_threads}") }, **opt_args)
+      else
+        Kernel.system(env, *cmd_args, **opt_args)
+      end
+    rescue RuntimeError => e
+      if modded_make_cmd && make_threads != 1
+        # retry with single thread if command is `make` and is modified by crew
+        warn "Command \"#{cmd_args.map { |arg| arg.sub('<<<CREW_NPROC>>>', "-j#{make_threads}") } .join(' ')}\" failed, retrying with \"-j1\"...".yellow
+        make_threads = 1
```

Comment:
What does setting this value do if it's not being used later?

Reply:
@supechicken Do you want to answer this so we can get it merged?

Reply:
Sorry for late reply😅 This variable will be reused in line 296-301 after calling `retry` (line 307) in this `rescue` part.

Reply:
OK, so this acts like a loop then.

Reply:
What can we do to prevent an endless loop?  Should we limit the number of retries?

Reply:
This retry logic will be skipped when `make_threads` is `1`

Reply:
@uberhacker Any other objections?

Reply:
Bump crew version?